### DISCUSSION TITLE
Testnet?

### DISCUSSION THREAD

**Discussion by tennox** [osmosis-labs/osmosis#588] [Q&A]
Is there a testnet available?

**Comment by daniel-farina** [comment]
Yes, there is a testnet available. We are currently working on new documentation for it. We will post it soon and let you know.

**Reply by gotjoshua** [reply]
Are permissionless smart contacts enabled on the testnet?

**Comment by czarcas7ic** [comment]
> Are permissionless smart contacts enabled on the testnet?

No, it is not. I cant speak for the devs, but I believe there will be some focus on utilizing the testnet more for testing out these upcoming features.

Also, wasnt the plan to make **permissioned** smart contracts via cosmwasm? Just my understanding through listening to updates from the lab but I could be wrong.

**Reply by ValarDragon** [reply]
The intended plan (dependent upon community approvals ofc) is for permissioned smart contracts on mainnet via cosmwasm.

The workflow for a testnet hasn't really been figured out. We probably do want the testnet to have effectively permissionless deployment, unclear how to achieve that and the goal of minimal testnet/mainnet state differences.
